util/example/tlgen/xbar_2x2.hjson

```diff
@@ -1,11 +1,48 @@
 {
   name: "2x2",
-  clock: "main",
+  clock: "clk_main_i",
+  clock_connections: {
+    clk_main_i: main
+    clk_periph_i: fixed
+  }
+  reset: "rst_main_ni",
+  reset_connections: {
+    rst_main_ni: sys
+    rst_periph_ni: sys_fixed
+  }
   nodes: [
-    { name: "h0", type: "host", clock: "main" },
-    { name: "h1", type: "host", clock: "periph" },
-    { name: "d0", type: "device", clock: "main", base_addr: "0x0000", size_byte: "0x1000" },
-    { name: "d1", type: "device", clock: "main", base_addr: "0x1000", size_byte: "0x1000" }
+    { name: "h0",
+      type: "host",
+      clock: "clk_main_i",
```

Comment:
do you think i should add an option in tlgen (maybe it's already there) to assume a default name/connection? That way we don't necessarily have to type this all out.

Reply:
Default clock for device/host is the main clock defined in xbar top.